### ISSUE TITLE
Kindle for Android - keys format changed

### ISSUE THREAD

**Issue by S-Graham** [apprenticeharper/DeDRM_tools#106]
I installed Kindle for Android 4.1 yesterday and downloaded a couple of the free books.

I've extracted AmazonSecureStorage.xml and it doesn't correspond with the code in androidkindlekey.py - "Could not retrieve Kindle for Android key."

I think the relevant lines are as follows:

```
<string name="AmazonSaltKeyV2">15700390cb7d18a3c41acfff0180cb50</string>
<string name="kindle.account.tokens">f4f72b042da87ab12b94e94a4dc16a027ac2f1cc</string>
<string name="AmazonKeyKeyV3">298c3e81d2335c5f02420ab527e05942e6064a7165c2e5e86d42e8b88af18389</string>

```
I'm running the code on Linux like this:
$ ./androidkindlekey.py -b AmazonSecureStorage.xml
and traced the code to see that function get_serials1 is entered, but that's as far as my debugging has got so far.

**Comment by apprenticeharper** [comment]
I haven't looked at Kindle for Android since Amazon turned off the ability to get the info with a simple backup. Is it possible to extract AmazonSecureStorage.xml without jail-breaking the Android device?

**Comment by S-Graham** [comment]
No. You need root access to copy the file, either by adb or to sd-card storage.

**Comment by apprenticeharper** [comment]
I shall see if I can take another look. It would be good to get Kindle for Android working again.

But if anyone else wants to have a go, I'd be delighted to accept a pull request.

**Comment by apprenticeharper** [comment]
It looks like someone needs to trace through the Kindle for Android app to see what it does with AmazonSaltKey2 and AmazonKeyKey3

**Comment by evg-zhabotinsky** [comment]
I confirm that `AmazonSecureStorage.xml` is useless as of now. However, `map_data_storage.db` still works like a charm. Tested it just now, with freshly installed Kindle4Android and Calibre plugin.

**Comment by apprenticeharper** [comment]
Thanks for the info. I just haven't got time to do anything with this info at present. Pull requests are welcome if anyone has working, tested, code.

**Comment by evg-zhabotinsky** [comment]
It's more like updating documentation to mention the possibility. I suspect that people who de-drm their books and those who root their phones overlap a lot. It's all about being in control of your property, after all.

**Comment by OdinVex** [comment]
Any chance we can create a k4a from `map_data_storage.db`? I've always ran with root on my devices and have the db.
